ts/features/bonus/siciliaVola/screens/voucherList/VoucherDetailsScreen.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from "react";
+import { useEffect } from "react";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { SafeAreaView, ScrollView, StyleSheet } from "react-native";
+import { View } from "native-base";
+import { fromNullable } from "fp-ts/lib/Option";
+import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
+import { emptyContextualHelp } from "../../../../../utils/emptyContextualHelp";
+import { IOStyles } from "../../../../../components/core/variables/IOStyles";
+import { H1 } from "../../../../../components/core/typography/H1";
+import { GlobalState } from "../../../../../store/reducers/types";
+import {
+  svGenerateVoucherBack,
+  svGenerateVoucherCompleted,
+  svGenerateVoucherGeneratedVoucher
+} from "../../store/actions/voucherGeneration";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import { VoucherRequest } from "../../types/SvVoucherRequest";
+import I18n from "../../../../../i18n";
+import { svVoucherDetailGet } from "../../store/actions/voucherList";
+import { SvVoucherId } from "../../types/SvVoucher";
+import {
+  selectedVoucherCodeSelector,
+  selectedVoucherSelector
+} from "../../store/reducers/selectedVoucher";
+import { H3 } from "../../../../../components/core/typography/H3";
+import CopyButtonComponent from "../../../../../components/CopyButtonComponent";
+import { isLoading, isReady } from "../../../bpd/model/RemoteValue";
+import { H4 } from "../../../../../components/core/typography/H4";
+import { formatDateAsLocal } from "../../../../../utils/dates";
+import { useIOBottomSheetRaw } from "../../../../../utils/bottomSheet";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { LoadingErrorComponent } from "../../../bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import VoucherDetailBottomSheet from "../../components/VoucherDetailBottomsheet";
+import { fromVoucherToDestinationLabels } from "../../utils";
+import { navigateToWorkunitGenericFailureScreen } from "../../../../../store/actions/navigation";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+const styles = StyleSheet.create({
+  itemRow: {
+    flex: 1,
+    flexDirection: "row",
+    justifyContent: "space-between"
+  }
+});
+
+const VoucherDetailsScreen = (props: Props): React.ReactElement | null => {
+  useEffect(() => {
+    if (props.selectedVoucherCode !== undefined) {
+      props.getVoucherDetail(props.selectedVoucherCode);
+    }
+  }, []);
```

Comment:
As the pr https://github.com/pagopa/io-app/pull/3397 was merged, now there is a warning
```
55:6  warning  React Hook useEffect has a missing dependency: 'props'. Either include it or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useEffect call and refer to those specific props inside useEffect  react-hooks/exhaustive-deps
```
Could you rewrite this hook in order to avoid the new warning?

Reply:
Yes, thank you! Updated in [84ab950](https://github.com/pagopa/io-app/pull/3400/commits/84ab950a1fa2d61df20356571ebdd76722474cdb)